raiden/raiden_event_handler.py

```diff
@@ -329,107 +353,148 @@ def handle_contract_send_channelunlock(
             raiden: RaidenService,
             channel_unlock_event: ContractSendChannelBatchUnlock,
     ):
+        token_network_identifier = channel_unlock_event.token_network_identifier
+        channel_identifier = channel_unlock_event.channel_identifier
         canonical_identifier = CanonicalIdentifier(
             chain_identifier=raiden.chain.network_id,
-            token_network_address=channel_unlock_event.token_network_identifier,
-            channel_identifier=channel_unlock_event.channel_identifier,
+            token_network_address=token_network_identifier,
+            channel_identifier=channel_identifier,
         )
         participant = channel_unlock_event.participant
         token_address = channel_unlock_event.token_address
-        triggered_by_block_hash = channel_unlock_event.triggered_by_block_hash
 
         payment_channel: PaymentChannel = raiden.chain.payment_channel(
             canonical_identifier=canonical_identifier,
         )
-        token_network: TokenNetwork = payment_channel.token_network
 
-        participants_details = token_network.detail_participants(
-            participant1=raiden.address,
-            participant2=participant,
-            block_identifier=triggered_by_block_hash,
-            channel_identifier=canonical_identifier.channel_identifier,
+        channel_state = get_channelstate_by_token_network_and_partner(
+            chain_state=state_from_raiden(raiden),
+            token_network_id=token_network_identifier,
+            partner_address=participant,
         )
 
-        our_details = participants_details.our_details
-        our_locksroot = our_details.locksroot
+        if not channel_state:
+            # channel was cleaned up already due to an unlock
+            return
 
-        partner_details = participants_details.partner_details
-        partner_locksroot = partner_details.locksroot
+        our_address = channel_state.our_state.address
+        our_locksroot = channel_state.our_state.onchain_locksroot
 
-        is_partner_unlock = (
-            partner_details.address == participant and
-            partner_locksroot != EMPTY_HASH
-        )
-        is_our_unlock = (
-            our_details.address == participant and
+        partner_address = channel_state.partner_state.address
+        partner_locksroot = channel_state.partner_state.onchain_locksroot
+
+        # we want to unlock because there are on-chain unlocked locks
+        search_events = (
```

Comment:
nitpick: move this back to one line? Same with the one below. I guess it fits in the 99 lines.